src/components/Calculator.tsx

```diff
@@ -0,0 +1,73 @@
+import Button from './Button';
+
+import useCalculator from 'hooks/useCalculator';
+
+import { OPERATION, DIGIT, MAX_LENGTH, ERROR_MESSAGE } from 'constant';
+import type { ValueOf } from 'types';
+
+type Digit = ValueOf<typeof DIGIT>;
+type Operation = ValueOf<typeof OPERATION>;
+
+function Calculator() {
+  const { calculator, calculate, clear, setOperand, setOperation } = useCalculator();
+  const { accumulator, augend, addend, operation } = calculator;
+
+  const handleDigitButton: React.MouseEventHandler<HTMLButtonElement> = (e) => {
+    if (!operation && String(augend).length >= MAX_LENGTH) {
+      alert(ERROR_MESSAGE.MAX_LENGTH);
+      return;
+    }
+
+    if (operation && String(addend).length >= MAX_LENGTH) {
+      alert(ERROR_MESSAGE.MAX_LENGTH);
+      return;
+    }
+
+    const digit = e.currentTarget.textContent as Digit;
```

Comment:
이럴 때는 alias보단 typeGuard 같은걸 사용해야 명확해지는 것 같아요!
```suggestion
    const digit = e.currentTarget.textContent;
    if (isDigit(digit)) {
      setOperand(digit);
    }
```
이러면 runtime에서도 type을 체크할 수 있어서 더 좋다고 생각합니다!

대신 isDigit 라는 함수를 구현해야겠죠?

Reply:
typeGuard가 있었네요 .. alias를 쓰면서도 만족스럽지 못했는데 isDigit이라는 typeGuard를 만들어서 활용하니 훨씬 좋은 것 같습니다. :)